### DISCUSSION TITLE
How to make logs Persistent ?

### DISCUSSION THREAD

**Discussion by ahussainrc** [cloudnative-pg/cloudnative-pg#4899] [Q&A]
CNPG relies on the PostgreSQL CSV log format, but it outputs logs on standard output in JSON format.
I understand that standard output JSON logs can be consumed by log aggregation tools like Loki/Grafana.
Currently, the log location is set to /controller/log, which is fixed and cannot be changed. Additionally, the /controller/log directory is not persistent.
How can I make the logs persistent so that I can find them if the log aggregation tool has not collected them for any reason?

**Comment by mnencia** [comment]
All the logs are converted into JSON and sent to the standard output. To persist them, you need something (i.e., promtail+loki, fluent bit, etc.) that gathers and stores them in a durable place.
See https://kubernetes.io/docs/concepts/cluster-administration/logging/ for more information on the Kubernetes Logging architecture.
